akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -175,27 +188,84 @@ private[akka] class Shard(typeName: String,
     None
   }
 
+  val lease = settings.leaseSettings.map(
+    ls ⇒
+      LeaseProvider(context.system).getLease(s"${context.system.name}-shard-$typeName-$shardId",
+                                             ls.leaseImplementation,
+                                             Cluster(context.system).selfAddress.hostPort))
+
+  val leaseRetryInterval = settings.leaseSettings match {
+    case Some(l) ⇒ l.leaseRetryInterval
+    case None ⇒ 5.seconds // not used
+  }
+
   initialized()
 
-  def initialized(): Unit = context.parent ! ShardInitialized(shardId)
+  def initialized(): Unit = {
```

Comment:
note that this is called from `recoveryCompleted` in the two implementations `DDataShard` and `PersistentShard` and those are doing more things including `become` which might interfere with this `awaitingLease`

Reply:
Ah this is a good point. I'll make the tetsts run with remembered entities and ddata and persistence

Reply:
ddata was indeed broken, pushed tests for that now